### PR TITLE
WIP revamp setting screen

### LINKED ISSUE #1937: [UX] Update settings page UI enhancement
**Name of feature to enhance**
There is no way of navigating out of the setting page in the current app thus need to add a top bar with navigation buttons 

**Description of feature**
Add top-bar navigation for the setting page and also some minor enhancements to rendering setting menu items 

**Describe the enhancement**

Default items

- [x] 1. Show top-bar navigation for the setting page
- [x] 2. Show user avatar
- [x] 3. Show a Logout button 
- [x] 4. Show FHIR Core Logo
- [x] 5. Show the App version and the last update if not available or configured on the Navigation drawer 

Configurable items

- [x] 1. Show a multilanguage support menu if configured 
- [x] 2. Show the App reset toggle*  - only show in app/debug mode 
- [x] 3. Show the P2P Sync menu* - Show this menu if [this](https://github.com/opensrp/fhir-resources/blob/d0b8c15a855aa6cc361f60d1a2ef69909ce027b2/ecbis_liberia/app_configurations/ecbis_cha_preview/navigation_config.json#L248-L263) is available and set to true.  
- [x] 4. Sync* - Will currently just trigger the sync. We can have this redirect to a different page in the future.

**Screenshot**

_NB. This is a stop-gap measure to have a better navigation UX for the settings page navigation. Final UI to be provided_

### PR DESCRIPTION
**Pull request by aurangzaibumer**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**
Fixes #1937 

Screenshot

Screenshot_1674036939

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by aurangzaibumer** [comment]
> Can you share a screenshot of the revamped UI?

Screenshot_1673851704

**Comment by aurangzaibumer** [comment]
Screenshot_1673864738
Screenshot_1673864742

cc @ellykits @f-odhiambo @dubdabasoduba

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/usersetting/UserSettingViewModel.kt [hunk 1]
```diff
@@ -116,4 +127,9 @@ constructor(
       accountAuthenticator.launchScreen(AppSettingActivity::class.java)
     }
   }
+
+  fun isP2PSyncAvailable(): Boolean {
+    val p2pMenu = (navigationConfiguration.staticMenu.find { it.id == "p2p_sync" })
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/usersetting/UserSettingViewModel.kt:132]: Can we add the `p2p_sync` to a constant so it can be reused.

File: android/engine/src/androidTest/java/org/smartregister/fhircore/engine/ui/components/UserSettingScreenKtTest.kt [hunk 2]
```diff
@@ -138,10 +139,12 @@ class UserSettingScreenKtTest {
           allowSwitchingLanguages = allowSwitchingLanguages,
           selectedLanguage = Locale.ENGLISH.toLanguageTag(),
           languages = listOf(Language("en", "English"), Language("sw", "Swahili")),
```
Comment (dubdabasoduba) [inline android/engine/src/androidTest/java/org/smartregister/fhircore/engine/ui/components/UserSettingScreenKtTest.kt:141]: Why do we have the languages hardcoded?

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: android/engine/src/androidTest/java/org/smartregister/fhircore/engine/ui/components/UserSettingScreenKtTest.kt (hunk 2 above)
Comment (aurangzaibumer) [reply android/engine/src/androidTest/java/org/smartregister/fhircore/engine/ui/components/UserSettingScreenKtTest.kt:148]: it's in the tests.

**Review by pld — DISMISSED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/usersetting/UserSettingViewModel.kt [hunk 3]
```diff
@@ -116,4 +129,9 @@ constructor(
       accountAuthenticator.launchScreen(AppSettingActivity::class.java)
     }
   }
+
+  fun isP2PSyncAvailable(): Boolean {
+    val p2pMenu = (navigationConfiguration.staticMenu.find { it.id == NAVIGATION_KEY_P2P })
+    return p2pMenu?.visible ?: false
+  }
 }
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/usersetting/UserSettingViewModel.kt:137]: Use the [ApplicationConfiguration.deviceToDeviceSync](https://github.com/opensrp/fhircore/blob/4d0c0abfc8216816d33bde1a4ea0cbacaaf5c64c/android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/app/ApplicationConfiguration.kt#L34) instead, like we discussed if the property is not null then p2p is enabled, disabled otherwise. We cannot rely on the navigation config.

**Review by ellykits — CHANGES_REQUESTED** [review]
